src/platforms/common/server/owning_c_str.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstdlib>
+#include <memory>
+
+namespace
+{
+class CStringFree
+{
+public:
+    void operator()(char* str)
+    {
+        if(str)
+        {
+            free(str);
+        }
+    }
+};
+}
+
+namespace mir
+{
+using CStr = std::unique_ptr<char[], CStringFree>;
+}
```

Comment:
That looks like a recipe for an ODR violation - thanks to the anon namespace `CStringFree` is a different type in each translation unit and that makes the `mir::CStr` definition different in each TU.